core-rust/state-manager/src/transaction/validator_transaction.rs

```diff
@@ -64,35 +67,35 @@ impl ValidatorTransaction {
         };
 
         PreparedValidatorTransaction {
-            hash: validator_role_nf_address,
-            manifest: TransactionManifest {
-                instructions: vec![instruction],
-                blobs: vec![],
-            },
+            hash,
+            instructions: vec![Instruction::System(instruction)],
+            blobs: vec![],
         }
     }
 }
 
 #[derive(Debug, Clone, TypeId, PartialEq, Eq)]
 pub struct PreparedValidatorTransaction {
     hash: Hash,
-    manifest: TransactionManifest,
+    instructions: Vec<Instruction>,
+    blobs: Vec<Vec<u8>>,
 }
 
 impl PreparedValidatorTransaction {
     pub fn get_executable(&self) -> Executable {
-        let transaction_hash = Hash([0u8; Hash::LENGTH]);
+        let transaction_hash = Hash([0u8; Hash::LENGTH]); // TODO: Is this okay?
```

Comment:
Hmm shouldn't we just use `self.hash` here?

Reply:
Yup, I think that should work.